src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -64,13 +64,22 @@ public function get($path)
 	 * Write the contents of a file.
 	 *
 	 * @param  string  $path
-	 * @param  string  $contents
+	 * @param  string|resource  $contents
 	 * @param  string  $visibility
 	 * @return bool
 	 */
 	public function put($path, $contents, $visibility = null)
 	{
-		return $this->driver->put($path, $contents, ['visibility' => $this->parseVisibility($visibility)]);
+		$config = ['visibility' => $this->parseVisibility($visibility)];
+        
+		if (is_resource($contents))
```

Comment:
no need for else

~~~~
                if (is_resource($contents))
		{
			return $this->driver->putStream($path, $contents, $config);
		}

 		return $this->driver->put($path, $contents, $config);
~~~~